python/lsst/pex/config/config.py

```diff
@@ -1382,6 +1404,43 @@ def compare(self, other, shortcut=True, rtol=1E-8, atol=1E-8, output=None):
         return compareConfigs(name, self, other, shortcut=shortcut,
                               rtol=rtol, atol=atol, output=output)
 
+    @classmethod
+    def __init_subclass__(cls, **kwargs):
+        """Run initialization for every subclass.
+
+        Specifically registers the subclass with a YAML representer
+        and YAML constructor (if pyyaml is available)
+        """
+        super().__init_subclass__(**kwargs)
+
+        if not yaml:
+            return
+
+        def config_representer(dumper, data):
+            """Represent a Config object in a form suitable for YAML.
+
+            Stores the serialized stream as a sequence of strings.
+            """
+            stream = io.StringIO()
+            data.saveToStream(stream)
+            configLines = stream.getvalue().split("\n")
+            typeString = _typeStr(type(data))
+            return dumper.represent_sequence(typeString, configLines)
+
+        def config_constructor(loader, node):
+            """Construct a config from YAML"""
+            configLines = loader.construct_sequence(node)
+            className = node.tag
+            cls = doImport(className)
```

Comment:
If the constructor is added to the loader, doesn't that mean that the class is already imported?

Reply:
Yes. I fixed it by using a generic label for writing out and including a test. We still need doImport though.